Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc

```diff
@@ -96,35 +96,43 @@ class HcalHBHEMuonAnalyzer : public edm::EDAnalyzer {
-  std::vector<bool>         innerTrack, OuterTrack, GlobalTrack;
-  std::vector<double>       IsolationR04,IsolationR03;
-  std::vector<double>       Energy,MuonHcalEnergy,MuonEcalEnergy,MuonHOEnergy;
-  std::vector<double>       MuonEcal3x3Energy,MuonHcal1x1Energy, Pmuon;
-  std::vector<unsigned int> MuonEcalDetId,MuonHcalDetId,MuonEHcalDetId;
-  std::vector<double>       MuonHcalHot,MuonHcalDepth1Energy;
-  std::vector<double>       MuonHcalDepth2Energy,MuonHcalDepth3Energy;
-  std::vector<double>       MuonHcalDepth4Energy,MuonHcalDepth5Energy;
-  std::vector<double>       MuonHcalDepth6Energy,MuonHcalDepth7Energy;
-  std::vector<double>       MuonHcalDepth1HotEnergy,MuonHcalDepth2HotEnergy;
-  std::vector<double>       MuonHcalDepth3HotEnergy,MuonHcalDepth4HotEnergy;
-  std::vector<double>       MuonHcalDepth5HotEnergy,MuonHcalDepth6HotEnergy;
-  std::vector<double>       MuonHcalDepth7HotEnergy,MuonHcalActiveLength;
+  std::vector<double>       ptGlob_, etaGlob_, phiGlob_, chiGlobal_;
+  std::vector<double>       globalMuonHits_,matchedStat_,globalTrckPt_;
+  std::vector<double>       globalTrckEta_,globalTrckPhi_,trackerLayer_;
+  std::vector<double>       innerTrackpt_,innerTracketa_,innerTrackphi_;
+  std::vector<double>       matchedId_,numPixelLayers_;
+  std::vector<double>       chiTracker_,dxyTracker_,dzTracker_;
+  std::vector<double>       outerTrackPt_,outerTrackEta_,outerTrackPhi_;
+  std::vector<double>       outerTrackChi_,outerTrackHits_,outerTrackRHits_;
+  std::vector<double>       tight_LongPara_,tight_PixelHits_,tight_TransImpara_;
+  std::vector<bool>         innerTrack_, outerTrack_, globalTrack_;
+  std::vector<double>       isolationR04_,isolationR03_;
+  std::vector<double>       energyMuon_,hcalEnergy_,ecalEnergy_,hoEnergy_;
+  std::vector<double>       ecal3x3Energy_,hcal1x1Energy_, pMuon_, hcalHot_;
+  std::vector<unsigned int> ecalDetId_,hcalDetId_,ehcalDetId_;
+  std::vector<double>       hcalDepth1Energy_, hcalDepth1ActiveLength_;
+  std::vector<double>       hcalDepth2Energy_, hcalDepth2ActiveLength_;
+  std::vector<double>       hcalDepth3Energy_, hcalDepth3ActiveLength_;
+  std::vector<double>       hcalDepth4Energy_, hcalDepth4ActiveLength_;
+  std::vector<double>       hcalDepth5Energy_, hcalDepth5ActiveLength_;
+  std::vector<double>       hcalDepth6Energy_, hcalDepth6ActiveLength_;
+  std::vector<double>       hcalDepth7Energy_, hcalDepth7ActiveLength_;
+  std::vector<double>       hcalDepth1EnergyHot_, hcalDepth1ActiveLengthHot_;
+  std::vector<double>       hcalDepth2EnergyHot_, hcalDepth2ActiveLengthHot_;
+  std::vector<double>       hcalDepth3EnergyHot_, hcalDepth3ActiveLengthHot_;
+  std::vector<double>       hcalDepth4EnergyHot_, hcalDepth4ActiveLengthHot_;
+  std::vector<double>       hcalDepth5EnergyHot_, hcalDepth5ActiveLengthHot_;
+  std::vector<double>       hcalDepth6EnergyHot_, hcalDepth6ActiveLengthHot_;
+  std::vector<double>       hcalDepth7EnergyHot_, hcalDepth7ActiveLengthHot_;
```

Comment:
I do not see how downstream code would need to know how the private data members are organized...
The branches written to the tree could still have the same name and would still be vectors.
You just replace seven lines of code with a loop containing one line of code (+1 line for the head of the loop).
Having this multiple times reduces a lot the complexity and makes reading the code a lot easier.